beacontools/scanner.py

```diff
@@ -100,6 +103,19 @@ def run(self):
             if event == LE_META_EVENT and subevent == EVT_LE_ADVERTISING_REPORT:
                 # we have an BLE advertisement
                 self.process_packet(pkt)
+        self.socket.close()
+
+    def set_scan_parameters(self, scan_type=ScanType.ACTIVE, interval_and_window_ms=(10, 10,),
+                            mac_type=BluetoothAddressType.RANDOM, filter_type=ScanFilter.ALL):
+        """"sets the scan parameters (socket, type, interval, window, own_type, filter)"""
+        from struct import pack
+        interval_ms, window_ms = interval_and_window_ms
+        scan_parameter_pkg = pack(
+            "<BBBBBBB",
```

Comment:
You should use the ">H" format character to encode the 16-bit values interval and window:
```
>>> struct.pack("<BB", 0, 0x10)
b'\x00\x10'
>>> struct.pack(">H", 0x10)
b'\x00\x10'
```
(This also allows to use values higher than 408 ms)

Reply:
okay, the new struct.pack format is now `>BHHBB`